app/admin/Settings.scala

```diff
@@ -17,19 +24,50 @@ case class Settings(
 )
 
 object Settings {
-  def fromConfig(config: Config): Settings =
-    Settings(
-      Switches(
-        PaymentMethodsSwitch.fromConfig(config.getConfig("oneOff")),
-        PaymentMethodsSwitch.fromConfig(config.getConfig("recurring")),
-        experimentsFromConfig(config, "abtests"),
-        SwitchState.fromConfig(config, "optimize"),
-        SwitchState.fromConfig(config, "internationalSubscribePages")
-      )
-    )
+  def fromDiskOrS3(config: Config, s3: AmazonS3): Either[Throwable, Settings] = {
+    val adminSettingsSource: Either[Throwable, AdminSettingsSource] = if (config.hasPath("local.path")) {
```

Comment:
Out of interest, what are your thoughts on always working with a global config vs passing around local bits of it?

Reply:
I'm happy with passing around local bits of it - but in this case it's poorly named (should be `adminSettingsSourceConfig` or similar)

Reply:
actually, thinking about this, maybe it's a bit silly to rely on the variable name to tell you which bit of the config you're in. if we always work on the global config then we don't need to worry about naming and we just assume that type `Config` means the whole config. was this what you were getting at?

Reply:
yeah, at the minor cost of  possibly longer path names, you get to reason about the config with no additional context